src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/AttachExceptionRecordToExistingCaseTest.java

```diff
@@ -136,7 +192,32 @@ private void invokeCallbackEndpointForLinkingDocsToCase(CaseDetails caseDetails,
             .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Bulk Scan Orchestrator Functional test")
             .body(callbackRequest)
             .when()
-            .post("/callback/attach_case");
+            .post("/callback/attach_case")
+            .jsonPath()
+            .getList("errors")
+            .isEmpty();
+    }
+
+    private Map<String, Object> exceptionRecordDataWithSearchFields(
+        CaseDetails targetCaseDetails,
+        CaseDetails exceptionRecord,
+        String searchCaseReferenceType
+    ) {
+        Map<String, Object> exceptionRecordData = new HashMap<>(exceptionRecord.getData());
+
+        if (searchCaseReferenceType == null) {
+            exceptionRecordData.put("attachToCaseReference", String.valueOf(targetCaseDetails.getId()));
+        } else {
+            exceptionRecordData.put("searchCaseReferenceType", searchCaseReferenceType);
+
+            String searchCaseReference = EXTERNAL_CASE_REFERENCE.equals(searchCaseReferenceType)
```

Comment:
`searchCaseReferenceType` is not null, so it's safe to compare variable to a constant the normal way.
Alternatively, there's always `Objects.equals(a, b)`

Reply:
Alternatives are just as good as current version. Is it the order that bothers you?

Reply:
to me it looks like `if (5 == x)` which I personally find quite weird

Reply:
`5 == x` would be weird, but `"CONSTANT".equals(var)` is pretty conventional. You're right it's not needed here, so I changed it.